### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by Sova171**

### CODE REVIEW

File: app/views/books/index.html.haml [hunk 1]
```diff
@@ -1,4 +1,24 @@
 %h1.card-title= t('.books')
+
+.dropdown
+  %button#dropdownMenuButton1.btn.dropdown-toggle{"aria-expanded" => "false", "data-bs-toggle" => "dropdown", type: "button"}
+    Categories
```
Comment (anastasiabalaban) [inline app/views/books/index.html.haml:5]: i18n

File: app/views/books/index.html.haml [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+  %ul.dropdown-menu{"aria-labelledby" => "dropdownMenuButton1"}
```
Comment (anastasiabalaban) [inline app/views/books/index.html.haml:6]: `{ "aria-labelledby": "dropdownMenuButton1" }` (add spaces)

File: app/facades/books/index_facade.rb [hunk 3]
```diff
@@ -19,14 +21,43 @@ def books
...
+
+    def ratings
+      Rating.all
+    end
+
     private
 
     def pagination
-      @pagination ||= pagy(all_books, page:)
+      @pagination ||= category.presence ? pagy_searchkick(all_books, page:) : pagy(all_books, page:)
```
Comment (anastasiabalaban) [inline app/facades/books/index_facade.rb:35]: you check `category.presence` in all_books method, do you also need to check this here?

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class BooksController < ApplicationController
   before_action :find_book, only: :show
 
   def index
-    @facade = ::Books::IndexFacade.new(page: params[:page])
+    @facade = ::Books::IndexFacade.new(page: params[:page], category: params[:category], rating: params[:rating])
```
[the remark this answers is not in the document]
Comment (Sova171) [reply app/controllers/books_controller.rb:7]: yes, rubocop checks the length of the line, but here the length is acceptable

[the remark this answers is not in the document]
Comment (Sova171) [reply app/controllers/books_controller.rb:7]: yes, rubocop checks the length of the line, but here the length is ok

**Pull request closed without merging**